src/main/scala/firrtl/transforms/FlattenRegUpdate.scala

```diff
@@ -7,11 +7,19 @@ import firrtl.ir._
 import firrtl.Mappers._
 import firrtl.Utils._
 import firrtl.options.Dependency
+import firrtl.InfoExpr.orElse
 
 import scala.collection.mutable
 
 object FlattenRegUpdate {
 
+  // Combination function for dealing with inlining of muxes and the handling of Triples of infos
+  private def combineInfos(muxInfo: Info, tinfo: Info, finfo: Info): Info = {
+    val (eninfo, tinfoAlt, finfoAlt) = MultiInfo.demux(muxInfo)
+    // Use MultiInfo constructor to preserve NoInfos
+    new MultiInfo(List(eninfo, orElse(tinfo, tinfoAlt), orElse(finfo, finfoAlt)))
```

Comment:
That is an interesting way to express it. I do feel like there is some other way to express this kind of concept but it smells like it might be a Haskell/ML thing that doesn't translate well to Scala. In any case, it's partially why I put it in a function since at least the composition is separated from the business logic of dealing with the muxes.